src/sessions.ts

```diff
@@ -355,10 +366,7 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
 
   /** Increment the transaction number on the internal ServerSession */
   incrementTransactionNumber(): void {
-    if (this.serverSession) {
-      this.serverSession.txnNumber =
-        typeof this.serverSession.txnNumber === 'number' ? this.serverSession.txnNumber + 1 : 0;
-    }
+    this[kTxnNumberIncrement] += 1;
```

Comment:
Just re-read this again, and I see we want to store the txnNumber on the session before acquiring the server session because we don't want to acquire the server session here. Can we update the description comment above explaining this?

Reply:
Updated!